lib/datadog/lambda/metrics.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+#
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+#
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2023 Datadog, Inc.
+#
+
+require 'datadog/statsd'
+
+module Datadog
+  # Metrics module contains the singleton class Client to send custom
+  # metrics to Datadog.
+  module Metrics
+    URL = 'localhost'
+    PORT = '8125'
+    # Client is a singleton class that instantiates a Datadog::Statsd
+    # client to send metrics to the Datadog Extension if present.
+    class Client
+      private_class_method :new
+
+      def self.instance
+        @instance ||= new
+      end
+
+      def distribution(name, value, time: nil, **tags)
+        tag_list = get_tags(**tags)
+
+        if Datadog::Utils.extension_running?
+          Datadog::Utils.logger.debug 'sending metrics through extension'
+          begin
+            @statsd.distribution(name, value, tags: tag_list)
+          rescue StandardError => e
+            Datadog::Utils.logger.warning "error sending metric to the extension: #{e}"
+          end
+        else
+          Datadog::Utils.logger.debug 'metrics are going to be handled by the forwarder'
+          time ||= Time.now
+          time_ms = time.to_f.to_i
+
+          metric = { e: time_ms, m: name, t: tag_list, v: value }
+          puts metric.to_json
+        end
+      end
+
+      def end
```

Comment:
I'd avoid `end` as a method name because it's a reserved keyword in Ruby.

It will technically work, but might create sharp edges: for example, if you call `end` from inside `Metrics::Client`, like 
```ruby
def another_method
  # do work
  end
end
```
the file won't compile.

Reply:
Interesting, thank you, will use another naming. Forgot it was reserved!